back/engines/commercial/texting/app/controllers/texting/web_api/v1/campaigns_controller.rb

```diff
@@ -49,6 +53,30 @@ def destroy
       head :ok
     end
 
+    def do_send
+      segments_limit = AppConfiguration.instance.settings('texting', 'monthly_sms_segments_limit')
+      if Texting::Sms.provider.exeeds_queue_limit?(@campaign.segments_count)
```

Comment:
Why do you sometimes use `@campaign` and other times `campaign`? We aren't using Rails views, but rather we are serializing to JSON, so I am just interested in what convention you are using, and why (to learn).

Reply:
Instance variable if `@campaign` is set in `before_action`.
Local variable if `campaign` is set in the action method.